### DISCUSSION TITLE
SDK Screen question

### DISCUSSION THREAD

**Discussion by JAlcaideH** [everysight-maverick/sdk#8] [General]
Hello,
I have connected the SDK using the sdk.key to the simulator, and it is already working. I have been looking at the examples as well. But I have a question. So If I would like to send a text to the screen or a Popup message when I need it from my application, should I create a new screen class and then destroy when I want to stop this message? Or maybe I haven't understand the onUpdateUI.
The thing is that a need to send messages to the glasses but only when I need it. What is the best way of doing it by using this SDK?
Than you, in advance.

**Comment by feranf** [comment]
Hi @JAlcaideH ,

The best way indeed is as you suggested: in the `onUpdateUI `function, add your `Text` to the screen using `Screen::add`. Keep the `Text` as a member of the class and remove it when needed via `Screen::remove`. The `Text` can also be modified while it is displayed.

Another option is to have the text added to the screen during its creation (`Screen:onCreate`), and modify its visibility when required (I personally prefer the first option).

Note that you may also use the `Screen::showPopup` function to display a `UIElement` like `Text` or a custom control you create (by inheriting from `UIElementGroup`).

Hope that helped.

**Reply by JAlcaideH** [reply]
First of all, thank for your answer.
Not sure If I have done as you said, this works for me but maybe there is something were I can improve by using the API. Also I hope this is useful for someone else.

I have something similar to this in my code. A mainScreen showing a message, and the classScreen to show an alert. Inside the classScreen I show an image within the onCreate method (Can we improve this?). Then in my main class, using the function showImg I can control which screen I am showing in each moment.

I will add some similar code below. (This is just an example related to my code).

```
Class screen:

class classScreen: Screen() {

    override fun onCreate() {

        val imageSrc = ImgSrc("myImage.png", ImgSrc.Slot.s0)

        val img = Image().setResource(imageSrc).setX(200f).setY(110f)
        add(img)
    }

}

My main class:

private var screenImg : classScreen? = null

private fun showImg() {
	if(!isShowing) {

		screenImg = ClassScreen()
		Evs.instance().screens().removeScreen(mainScreen!!)
                Evs.instance().screens().addScreen(screenImg!!)

	} else {

		Evs.instance().screens().removeScreen(screenImg!!)
                Evs.instance().screens().addScreen(mainScreen!!)
	}
}

```
Also, I would like to ask. Is it possible to set a time through the sdk in order to stop showing this image? or otherwhise I can only do this through a popUpMessage.
To be honest, I still don't know how the onUpdateUI works.

**Reply by feranf** [reply]
Hi,

That's a good question. I've modified the [Android ImageHandling](https://github.com/everysight-maverick/samples/tree/main/android/ImagesHandling) sample to include a showcase that might answer your question. I've implemented some dynamic screen logic within the [ImagePopupScreen.kt](https://github.com/everysight-maverick/samples/blob/main/android/ImagesHandling/src/main/java/com/everysight/sdk/samples/imageshandling/ImagePopupScreen.kt) class.

The next SDK version will also include an option to add a UIElement for a certain period of time.

Hope that helped.

**Comment by feranf** [comment]
Hi,
Another point that might help: you can think of:

- a `Screen` as an Android `Activity`/`Fragment` 
- `UIElement` along with its inheritors as Android `Views`.
- `UIElementGroup` as Android `ViewGroup`
